CSVSource/src/main/java/io/vantiq/extsrc/CSVSource/CSVReader.java

```diff
@@ -47,44 +62,171 @@ static void sendNotification(String filename, int numPacket, ArrayList<Map<Strin
+            if (schema.get(field) != null) {
+                field = schema.get(field);
             }
         }
-        return field ; 
+        return field;
+    }
+
+    static Map<String, FixedRecordfieldInfo> fixedRecord(Map<String, Map<String, String>> schema) {
+
+        Map<String, FixedRecordfieldInfo> recordInfo = new HashMap<String, FixedRecordfieldInfo>();
+        Set<String> fieldList = schema.keySet();
+        int recordSize = 0;
+
+        for (String field : fieldList) {
+            Map<String, String> fieldInf = schema.get(field);
+            FixedRecordfieldInfo o = FixedRecordfieldInfo.Create(fieldInf);
+
+            recordInfo.put(field, o);
+
+        }
+        return recordInfo;
+    }
+
+    static int fixedRecordLength(Map<String, FixedRecordfieldInfo> schema) {
+
+        Set<String> fieldList = schema.keySet();
+        int recordSize = 0;
+
+        for (String field : fieldList) {
+            FixedRecordfieldInfo o = schema.get(field);
+
+            if (recordSize < o.offset + o.length) {
+                recordSize = o.offset + o.length;
+            }
+
+        }
+        return recordSize + 1;// CR+LF
     }
 
     /**
-     * Responsible for reading the lines from the file and converting it events to be sent to server. 
-     * Each line is split by the delimiter and then, based on the schema object, determine the 
-     * attribute name. 
+     * Responsible for reading records from the file and converting to events to be
+     * sent to server. Each record is a fixed record and, based on the schema
+     * object, we extract the field.
+     * 
      * @param csvFile
      * @param config
      * @param oClient
      * @return
+     * @throws InterruptedException
      */
     @SuppressWarnings("unchecked")
-    static public ArrayList<Map<String,String>> execute(String csvFile, Map<String, Object> config, ExtensionWebSocketClient oClient) {
+    static public ArrayList<Map<String, String>> executeFixedRecord(String csvFile, Map<String, Object> config,
+            ExtensionWebSocketClient oClient) throws InterruptedException {
+
+        int numOfRecords; // This is the total number of records/lines processed from the file.
+        int packetIndex = 0;
+        Map<String, Map<String, String>> schema = null;
+        Map<String, FixedRecordfieldInfo> recordMetaData = null;
+
+        if (config.get("schema") != null) {
+            schema = (Map<String, Map<String, String>>) config.get("schema");
+        }
+
+        recordMetaData = fixedRecord(schema);
+
+        int recordSize = fixedRecordLength(recordMetaData);
+        if (config.get("fixedRecordSize") != null)
+            recordSize = (int) config.get("fixedRecordSize");
+
+        int MaxLinesInEvent = (int) config.get(MAX_LINES_IN_EVENT);
+
+        int SleepBetweenPackets = 0;
+        if (config.get("waitBetweenTx") != null)
+            SleepBetweenPackets = (int) config.get("waitBetweenTx");
```

Comment:
```suggestion
            sleepBetweenPackets = (int) config.get("waitBetweenTx");
```

This is required? handle nulls? or zero?  Check for negative?

Reply:
it helped to verify some stuff during the development.